### PR TITLE
CA-197 | Add generalized data elements

### PR DESCRIPTION
**Pull request by sriram-privado**
Jira issue - https://privado-ai.atlassian.net/browse/CA-197

### CODE REVIEW

File: rules/sources/contact_data.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ sources:
     isSensitive: False
     sensitivity: medium
     patterns:
-      - "(?i)((phone|mobile|cellphone|contact)|(.*((phone|mobile|cellphone|contact)[^\\s/(;)#|,=!>]{0,2}(?:detail|details)|(?:mobile|phone|cellphone|contact)[^\\s/(;)#|,=!>]{0,10}(?:number|nbr|num|no)|(?:primary|secondary)[^\\s/(;)#|,=!>]{0,10}(phone|landline|telephone))))"
+      - "(?i)((phone|mobile|cellphone|contact|work)|(.*((phone|mobile|cellphone|contact)[^\\s/(;)#|,=!>]{0,2}(?:detail|details)|(?:mobile|phone|cellphone|contact)[^\\s/(;)#|,=!>]{0,10}(?:number|nbr|num|no)|(?:primary|secondary)[^\\s/(;)#|,=!>]{0,10}(phone|landline|telephone))))"
```
Comment (khemrajrathore) [inline rules/sources/contact_data.yaml:28]: Do we intend to match `work` in  `ContactData.Phonenumber`?

File: rules/sources/online_identifiers.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ sources:
     isSensitive: False
     sensitivity: medium
     patterns:
-      - "(?i).*(device[^\\s/(;)#|,=!>]{0,5}(?:id|info|details|token))|imei"
+      - "(?i).*(device[^\\s/(;)#|,=!>]{0,5}(?:id|info|details|token))|imei|fingerprint"
```
Comment (khemrajrathore) [inline rules/sources/online_identifiers.yaml:18]: I guess `fingerprint` was there earlier, but was resulting in FP, @HiteshMah-Jan can confirm

File: rules/sources/contact_data.yaml (hunk 1 above)
Comment (sriram-privado) [reply rules/sources/contact_data.yaml:28]: Only 'workPhone' , this wasn't getting matched by previous regex.

Comment (sriram-privado) [reply rules/sources/contact_data.yaml:28]: @khemrajrathore should I change this regex to be something else ?

File: rules/sources/online_identifiers.yaml (hunk 2 above)
Comment (sriram-privado) [reply rules/sources/online_identifiers.yaml:18]: I can remove this if thats the case. @HiteshMah-Jan @khemrajrathore

File: rules/sources/contact_data.yaml (hunk 1 above)
Comment (khemrajrathore) [reply rules/sources/contact_data.yaml:28]: If we intent to match `workPhone` update the regex from `work` to `workPhone` as with current regex `workphone` won't get matched unless we explicitly add a `.*`

Comment (sriram-privado) [reply rules/sources/contact_data.yaml:28]: With current regex, `workPhone` does match, can you take a look again ?

File: rules/sources/online_identifiers.yaml (hunk 2 above)
Comment (sriram-privado) [reply rules/sources/online_identifiers.yaml:18]: Let me know if I should remove this.

**Comment by sriram-privado** [comment]
Closing this PR since 'workphone' rule is already merged. Also, `fingerprint` might lead to more false-positives as mentioned in the comment.

**Pull request closed without merging**